nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -105,33 +110,50 @@ boolean isConnected() {
         return socket != null && socket.isConnected();
     }
 
+    private InetSocketAddress inetSocketAddress() {
+        DnsResolver dnsResolver = connectionKey.dnsResolver();
+        if (dnsResolver.useDefaultJavaResolver()) {
+            return new InetSocketAddress(connectionKey.host(), connectionKey.port());
+        } else {
+            InetAddress address = dnsResolver.resolveAddress(connectionKey.host(), connectionKey.dnsAddressLookup());
+            return new InetSocketAddress(address, connectionKey.port());
+        }
+    }
+
+    private int connectToProxy(InetSocketAddress remoteAddress) throws IOException {
+        HelidonSocket tempSocket = PlainSocket.client(socket, createChannelId(socket));
+        StringBuilder httpConnect = new StringBuilder();
+        httpConnect.append("CONNECT ").append(remoteAddress.getHostName())
+        .append(":").append(remoteAddress.getPort()).append(" HTTP/1.1\r\n")
+        .append("Host: ").append(remoteAddress.getHostName()).append(":").append(remoteAddress.getPort())
+        .append("\r\n")
+        .append("Accept: */*\r\n\r\n");
```

Comment:
This is the part I would like to rewrite with my comment of here: https://github.com/helidon-io/helidon/pull/6441#discussion_r1200122792

In case we go forward with this, I will make it as you suggest.